### DISCUSSION TITLE
Add new plots in parallel automatically?

### DISCUSSION THREAD

**Discussion by lilei105** [Chia-Network/chia-blockchain#1912] [Q&A]
I'm using a 1T SSD as cache. With v1.04 software I was able to run 4 plots in parallel (6min delay) successfully. Well my CPU is kind of old so it took 12 hours to finish. I hate to set an alarm clock to get up at 3AM to start 4 new plots. I know there is a plot queue. But the headache is, the queue only runs 1 plot, which is a huge waste. How can I run 4 plots in parallel continuously? In a ideal way I'm expecting if any plot finished successfully, 1 new plot would start automatically, thus there are N plots always running at any given time.

**Comment by berlef80** [comment]
name your plots in advanced options and queue each one up.  The differently named plots will plot in parrallel.  I have found if you queue more than 3-4 it crashes at some point, but it will at least get you through the nights.

**Reply by padik11** [reply]
That doesn't solve the case. Now on 5 ssd, I can draw 20 plots in parallel, but I have to get up at night. If I queue it in series, I would have to have 20 SSD.

**Reply by XIII39** [reply]
I assume you're using the GUI. Don't set it up as parallel plots, set it up as plot in queue

Say you have a 2tb SSD and you want to make 100 plots, let's assume you are making 4 plots in parallel.
Create a plot in queue
Set your plot count to 25
Name the queue SSD1_A
tmp drive is your SSD

Once you've created that, make a second one 
Set your plot count to 25
Name the queue SSD1_B
tmp drive is your SSD

Then a third
Set your plot count to 25
Name the queue SSD1_C
tmp drive is your SSD

Finally a forth
Set your plot count to 25
Name the queue SSD1_D
tmp drive is your SSD

Here you will be creating 100 plots on the single SSD, essentially plotting 4 in parallel

**Reply by padik11** [reply]
It makes sense, but I will have to set these tasks every 15-25 minutes anyway, because this is what I need to write to the HDD and I do not want to save at the same time. Unless I buy 20 HDDs, but it's still a good solution.

**Comment by cciollaro** [comment]
agreed, using the queues are probably the next step.

if you're using CLI though and I guess non-windows (although I imagine this will work on windows soon) you can check out https://github.com/ericaltendorf/plotman. this acts as an orchestration layer and can run your plotting commands in a loop, make sure you always have the right number of plots running, and perform staggering so that the plotting jobs are running on different phases.

**Comment by lilei105** [comment]
Thanks a lot, guys. Now I'm able to write my own CLI script. Works great.

**Reply by gunshit** [reply]
> Thanks a lot, guys. Now I'm able to write my own CLI script. Works great.

Noice! could you please share it? ^_^
